### ISSUE TITLE
DIGITS has different output to Caffe

### ISSUE THREAD

**Issue by RadicoLabs** [NVIDIA/DIGITS#411]
Hey people,

So let me be the first to say that i have read through as much as i can on google, including this somewhat useful thread, linked [here](https://groups.google.com/forum/#!topic/digits-users/eRGijUu9q30), but dont know the necessary steps to implement this fix.

We are attempting to build a somewhat basic nudity scanner in Caffe/DIGITS, and so far our progress is good, except for with one of our models, DIGITS classifies an image as 99% nudity, which is correct. However, when using the Python wrapper [use_archive.py](https://github.com/NVIDIA/DIGITS/blob/master/examples/classification/use_archive.py) and [example.py](https://github.com/NVIDIA/DIGITS/blob/master/examples/classification/example.py), Caffe gives almost the exact opposite output, classifying the image as 99.99% non-nudity. 

In an attempt to follow the aforementioned thread, i have turned off oversampling in the Python/caffe/Classifier.py script and rebuilt the library however caffe is still outputting the wrong result.

I should probably point out that i am using images scaled to 256x256... So i dont know why there seems to be a problem

**Comment by lukeyeager** [comment]
Can you get the results to match up perfectly for other images? Is it just one image that's entirely wrong?

Are you classifying multiple images at once and running into this bug that @yangj1e just fixed? #409

Are you subtracting the mean image or mean pixel? DIGITS may handle the order of resize/crop/subtract differently for mean files than Caffe does.

**Comment by RadicoLabs** [comment]
Okay so after some more testing, we found that whilst DIGITS classifies an image with 99% confidence, Caffe will still classify it, just with a much small confidence margin (approx 5-25% depending on image).

Will try the 409 fix soon though and get back to you!

**Comment by lukeyeager** [comment]
Can you post some screenshots of the different output you're seeing? I still don't understand whether it's only one image that has a problem or the entire set of test images.

**Comment by lukeyeager** [comment]
After re-reading your OP, maybe posting screenshots isn't the best idea after all :wink:.

Did you figure it out? If so, could you let us know what you had to change? Classification results not matching between DIGITS and pycaffe is a recurring issue that I would like to do a better job of documenting.

**Comment by bnascimento** [comment]
Guys, im also having the same problem.
LeNet classification models trained under DIGITS perform slightly worst under a windows caffe port.
Good news is that neither the configuration or platform makes a difference. I mean, I trained the models under Digits using the GPU and under x64. If I use this models on the windows caffe port, using the GPU or the CPU on 64 or 32 bits makes absolutely no difference.
Anyone else got the same problems?

**Comment by RadicoLabs** [comment]
@lukeyeager

Sorry for the long awaited reply, Been away for a while.... Im going to be playing around with DIGITS over the coming weeks so I will be able to check up on this error again :)

**Comment by lukeyeager** [comment]
@RadicoLabs ok great. I've updated the list of "Limitations" with the classification example. Take a look in case it's helpful:
https://github.com/NVIDIA/DIGITS/blob/cb0cba0e1f/examples/classification/README.md#limitations

**Comment by kielnino** [comment]
I have the same problem. I've trained a GoogLeNet with Pixel-wise mean and the _Squash_ Resize Transformation. Then I used the example https://github.com/NVIDIA/DIGITS/blob/master/examples/classification/example.py to classify.

If I classify a test-image with the classify-one-button in DIGITS I get:

- 84.13% (Class "1")
- 15.87% (Class "0")

But in Python I get:

- 86.6031% (Class "1")
- 13.3965% (Class "0")

The difference is small. But in other cases it is even worse...
I'm using DIGITS 3.0.0 from the Ubuntu Repository.
